astropy/io/fits/tests/test_fsspec.py

```diff
@@ -0,0 +1,123 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Can `astropy.io.fits.open` access (remote) data using the fsspec package?
+"""
+from astropy.io import fits
+from astropy.nddata import Cutout2D
+from astropy.utils.compat.optional_deps import HAS_FSSPEC, HAS_S3FS  # noqa
+from astropy.utils.data import get_pkg_data_filename, conf
+
+import numpy as np
+from numpy.testing import assert_array_equal, assert_allclose
+import pytest
+
+if HAS_FSSPEC:
+    import fsspec
+
+
+@pytest.mark.skipif("not HAS_FSSPEC")
+def test_fsspec_local():
+    """Can we use fsspec to read a local file?"""
+    fn = get_pkg_data_filename('data/test0.fits')
+    hdulist_classic = fits.open(fn)
+    hdulist_fsspec = fits.open(fn, use_fsspec=True)
+    assert_array_equal(hdulist_classic[2].data, hdulist_fsspec[2].data)
+    assert_array_equal(hdulist_classic[2].section[3:5], hdulist_fsspec[2].section[3:5])
+    hdulist_classic.close()
+    hdulist_fsspec.close()
+
+
+@pytest.mark.skipif("not HAS_FSSPEC")
+def test_fsspec_local_write(tmpdir):
+    """Can we write to a local file that was opened using fsspec?"""
+    fn = get_pkg_data_filename('data/test0.fits')
+    fn_tmp = tmpdir / "tmp.fits"
+    with fits.open(fn, use_fsspec=True) as hdul:
+        # writing to a section is never allowed
+        with pytest.raises(TypeError):
+            hdul[1].section[0, 0] = -999
+        # however writing to .data should work
+        hdul[1].data[2, 3] = -999
+        assert hdul[1].data[2,3] == -999
+        hdul.writeto(fn_tmp)
+
+    # Is the new value present when we re-open the file?
+    with fits.open(fn_tmp) as hdul:
+        assert hdul[1].data[2,3] == -999
+
+
+@pytest.mark.remote_data
+@pytest.mark.skipif("not HAS_FSSPEC")
+def test_fsspec_http():
+    """Can we use fsspec to open a remote FITS file via http?"""
+    uri = "https://mast.stsci.edu/api/v0.1/Download/file/?uri=mast:HST/product/j8pu0y010_drc.fits"
+    # Expected array was obtained by downloading the file locally and executing:
+    # with fits.open(local_path) as hdul:
+    #     hdul[1].data[1000:1002, 2000:2003]
+    expected = np.array([[0.00600598, 0.00546154, 0.00134362],
```

Comment:
This is a good point.  Does reprocessing happen frequently?  I will look around for raw data -- that is an excellent idea.

Reply:
Maybe you can check some keys in the header which shouldn't change? Also, that would make for a smaller download.

Reply:
Reprocessing happens as needed, so frequent enough, because there is always better calibration files.

Reply:
I believe I fixed this issue by modifying the tests to access a Hubble jitter FITS file (*.jif) rather than a real image.  The jif file I selected does not appear to have changed since the year the observations were made (2012), i.e., it looks like jitter files are not affected by reprossing?

Commit: https://github.com/astropy/astropy/pull/13238/commits/5eaae6a5bb63cf884ee90c2637bc4099b5e14053